test-network-function/accesscontrol/suite.go

```diff
@@ -264,24 +264,80 @@ func testRoles(env *config.TestEnvironment) {
 	testServiceAccount(env)
 	testRoleBindings(env)
 	testClusterRoleBindings(env)
+	testAutomountService(env)
 }
 
 func testServiceAccount(env *config.TestEnvironment) {
 	testID := identifiers.XformToGinkgoItIdentifier(identifiers.TestPodServiceAccountBestPracticesIdentifier)
 	ginkgo.It(testID, func() {
 		ginkgo.By("Should have a valid ServiceAccount name")
 		for _, podUnderTest := range env.PodsUnderTest {
+			ginkgo.By(fmt.Sprintf("Testing pod service account %s %s", podUnderTest.Namespace, podUnderTest.Name))
+			serviceAccountName := podUnderTest.ServiceAccount
+			gomega.Expect(serviceAccountName).ToNot(gomega.BeEmpty())
+		}
+	})
+}
+
+//nolint:funlen
+func testAutomountService(env *config.TestEnvironment) {
+	testID := identifiers.XformToGinkgoItIdentifier(identifiers.TestPodAutomountServiceAccountIdentifier)
+	ginkgo.It(testID, func() {
+		ginkgo.By("Should have automountServiceAccountToken set to false")
+		msg := []string{}
+		for _, podUnderTest := range env.PodsUnderTest {
+			ginkgo.By(fmt.Sprintf("check the existence of pod service account %s %s", podUnderTest.Namespace, podUnderTest.Name))
```

Comment:
Minor fix for readability: "... service account %s %s" -> "service account for pod %s (ns %s), podUnderTest.Name, podUnderTest.Namespace"

Reply:
fixed